### ISSUE TITLE
global-default-backend does not find services in K8S

### ISSUE THREAD

**Issue by prg3** [containous/traefik#4465]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Added service and ingress as below configuration.

### What did you expect to see?

On the dashboard I would expect to see the "global-default-backend" to show green and have the backend services.

### What did you see instead?

global-default-backend shows red and no services attached.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.7.8
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

ds.yaml
```yaml
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: traefik-ingress-controller
  namespace: kube-system
---
kind: DaemonSet
apiVersion: extensions/v1beta1
metadata:
  name: traefik-ingress-controller
  namespace: kube-system
  labels:
    k8s-app: traefik-ingress-lb
    app: traefik
spec:
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb
        name: traefik-ingress-lb
    spec:
      serviceAccountName: traefik-ingress-controller
      terminationGracePeriodSeconds: 60
      containers:
      - image: traefik
        name: traefik-ingress-lb
        ports:
        - name: http
          containerPort: 80
          hostPort: 80
        - name: admin
          containerPort: 8080
          hostPort: 8080
        securityContext:
          capabilities:
            drop:
            - ALL
            add:
            - NET_BIND_SERVICE
        args:
        - --api
        - --kubernetes
        - --logLevel=DEBUG
        - --sendAnonymousUsage=true
        - --metrics.prometheus=true
        - --accesslog.format=json
---
kind: Service
apiVersion: v1
metadata:
  name: traefik-ingress-service
  namespace: kube-system
spec:
  type: NodePort
  selector:
    k8s-app: traefik-ingress-lb
  ports:
    - protocol: TCP
      port: 80
      nodePort: 30080
      name: web
    - protocol: TCP
      port: 8080
      nodePort: 30088
      name: admin
ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: missing
spec:
  backend:
    serviceName: missing
    servicePort: 80
```
pod.json
```son
{
    "apiVersion": "extensions/v1beta1",
    "kind": "Deployment",
    "metadata": {
        "name": "missing"
    },
    "spec": {
        "replicas": 1,
        "template": {
            "metadata": {
                "labels": {
                    "app": "missing",
                    "name": "missing"
                },
                "name": "missing"
            },
            "spec": {
                "containers": [
                    {
                        "image": "nginxdemos/hello",
                        "name": "app",
                        "ports": [
                            {
                                "containerPort": 80,
                                "name": "http"
                            }
                        ]
                    }
                ]
	}
        }
    }
}
```
service.json
```son
{
  "apiVersion": "v1",
  "metadata": {
    "name": "missing"
  },
  "kind": "Service",
  "spec": {
    "ports": [
      {
        "targetPort": 80,
        "protocol": "TCP",
        "port": 80,
        "name": "http"
      }
    ],
    "selector": {
      "app": "missing"
    }
  }
}
```
ui-ict.yaml
```yaml
---
apiVersion: v1
kind: Service
metadata:
  name: traefik-web-ui
  namespace: kube-system
spec:
  selector:
    k8s-app: traefik-ingress-lb
  ports:
  - name: web
    port: 80
    targetPort: 8080
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: traefik-web-ui
  namespace: kube-system
spec:
  rules:
  - host: dashboard.example.com
    http:
      paths:
      - path: /
        backend:
          serviceName: traefik-web-ui
          servicePort: web
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
time="2019-02-04T20:53:40Z" level=info msg="Traefik version v1.7.8 built on 2019-01-29_04:33:36PM"
time="2019-02-04T20:53:40Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":false,\"CheckNewVersion\":true,\"SendAnonymousUsage\":true,\"AccessLogsFile\":\"\",\"AccessLog\":{\"format\":\"json\"},\"TraefikLogsFile\":\"\",\"TraefikLog\":null,\"Tracing\":null,\"LogLevel\":\"DEBUG\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"traefik\":{\"Address\":\":8080\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":null,\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null},\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":{\"EntryPoint\":\"traefik\",\"Dashboard\":true,\"Debug\":false,\"CurrentConfigurations\":null,\"Statistics\":null},\"Metrics\":{\"Prometheus\":{\"Buckets\":[0.1,0.3,1.2,5],\"EntryPoint\":\"traefik\"},\"Datadog\":null,\"StatsD\":null,\"InfluxDB\":null},\"Ping\":null,\"HostResolver\":null}"
time="2019-02-04T20:53:40Z" level=info msg="\nStats collection is enabled.\nMany thanks for contributing to Traefik's improvement by allowing us to receive anonymous information from your configuration.\nHelp us improve Traefik by leaving this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2019-02-04T20:53:40Z" level=debug msg="Configured Prometheus metrics"
time="2019-02-04T20:53:40Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc00043a5e0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2019-02-04T20:53:40Z" level=info msg="Preparing server traefik &{Address::8080 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc00043a600} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2019-02-04T20:53:40Z" level=info msg="Starting server on :80"
time="2019-02-04T20:53:40Z" level=info msg="Starting server on :8080"
time="2019-02-04T20:53:40Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2019-02-04T20:53:40Z" level=info msg="Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null}"
time="2019-02-04T20:53:40Z" level=debug msg="Using Ingress label selector: \"\""
time="2019-02-04T20:53:40Z" level=info msg="ingress label selector is: \"\""
time="2019-02-04T20:53:40Z" level=info msg="Creating in-cluster Provider client"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Service"
time="2019-02-04T20:53:40Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2019-02-04T20:53:40Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"global-default-backend\":{\"loadBalancer\":{\"method\":\"wrr\"}},\"dashboard.example.com/\":{\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"global-default-frontend\":{\"entryPoints\":[\"http\"],\"backend\":\"global-default-backend\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null},\"dashboard.example.com/\":{\"entryPoints\":[\"http\"],\"backend\":\"dashboard.example.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"dashboard.example.com\":{\"rule\":\"Host:dashboard.example.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2019-02-04T20:53:40Z" level=debug msg="Wiring frontend global-default-frontend to entryPoint http"
time="2019-02-04T20:53:40Z" level=debug msg="Creating backend global-default-backend"
time="2019-02-04T20:53:40Z" level=debug msg="Adding TLSClientHeaders middleware for frontend global-default-frontend"
time="2019-02-04T20:53:40Z" level=debug msg="Creating load-balancer wrr"
time="2019-02-04T20:53:40Z" level=debug msg="Creating route / PathPrefix:/"
time="2019-02-04T20:53:40Z" level=debug msg="Wiring frontend dashboard.example.com/ to entryPoint http"
time="2019-02-04T20:53:40Z" level=debug msg="Creating backend dashboard.example.com/"
time="2019-02-04T20:53:40Z" level=debug msg="Adding TLSClientHeaders middleware for frontend dashboard.example.com/"
time="2019-02-04T20:53:40Z" level=debug msg="Creating load-balancer wrr"
time="2019-02-04T20:53:40Z" level=debug msg="Creating route dashboard.example.com Host:dashboard.example.com"
time="2019-02-04T20:53:40Z" level=debug msg="Creating route / PathPrefix:/"
time="2019-02-04T20:53:40Z" level=info msg="Server configuration reloaded on :8080"
time="2019-02-04T20:53:40Z" level=info msg="Server configuration reloaded on :80"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Secret"
time="2019-02-04T20:53:40Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2019-02-04T20:53:40Z" level=debug msg="Skipping Kubernetes event kind *v1.Secret"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Secret"
time="2019-02-04T20:53:40Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2019-02-04T20:53:40Z" level=debug msg="Skipping Kubernetes event kind *v1.Secret"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:40Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:40Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"global-default-backend\":{\"loadBalancer\":{\"method\":\"wrr\"}},\"dashboard.example.com/\":{\"servers\":{\"traefik-ingress-controller-99qvr\":{\"url\":\"http://10.1.0.9:8080\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"global-default-frontend\":{\"entryPoints\":[\"http\"],\"backend\":\"global-default-backend\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null},\"dashboard.example.com/\":{\"entryPoints\":[\"http\"],\"backend\":\"dashboard.example.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"dashboard.example.com\":{\"rule\":\"Host:dashboard.example.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2019-02-04T20:53:40Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:40Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:41Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:41Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:42Z" level=debug msg="Wiring frontend global-default-frontend to entryPoint http"
time="2019-02-04T20:53:42Z" level=debug msg="Creating backend global-default-backend"
time="2019-02-04T20:53:42Z" level=debug msg="Adding TLSClientHeaders middleware for frontend global-default-frontend"
time="2019-02-04T20:53:42Z" level=debug msg="Creating load-balancer wrr"
time="2019-02-04T20:53:42Z" level=debug msg="Creating route / PathPrefix:/"
time="2019-02-04T20:53:42Z" level=debug msg="Wiring frontend dashboard.example.com/ to entryPoint http"
time="2019-02-04T20:53:42Z" level=debug msg="Creating backend dashboard.example.com/"
time="2019-02-04T20:53:42Z" level=debug msg="Adding TLSClientHeaders middleware for frontend dashboard.example.com/"
time="2019-02-04T20:53:42Z" level=debug msg="Creating load-balancer wrr"
time="2019-02-04T20:53:42Z" level=debug msg="Creating server traefik-ingress-controller-99qvr at http://10.1.0.9:8080 with weight 1"
time="2019-02-04T20:53:42Z" level=debug msg="Creating route / PathPrefix:/"
time="2019-02-04T20:53:42Z" level=debug msg="Creating route dashboard.example.com Host:dashboard.example.com"
time="2019-02-04T20:53:42Z" level=info msg="Server configuration reloaded on :80"
time="2019-02-04T20:53:42Z" level=info msg="Server configuration reloaded on :8080"
time="2019-02-04T20:53:42Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:42Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2019-02-04T20:53:43Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"```

**Comment by prg3** [comment]
Building off of the patch branch solves the problem on my test system.

**Comment by traefiker** [comment]
Closed by #4486.
